### PR TITLE
[CI] Fix pulling of new updates in docker build

### PR DESCRIPTION
**Pull request by Tankilevitch**
1. In Dockerfile, changed to use apt-get upgrade because the filter we used to catch security updates didn't catch all security updates.
2. In Makefile add another validation for running `docker pull` and `--cache-from` which will add those commands only if `MLRUN_NO_CACHE` isn't defined. 
3. Removed `cache_from_tag` from `build.yaml` as this is no more needed because makefile can handle both `MLRUN_NO_CACHE` and `MLRUN_DOCKER_CACHE_FROM_TAG`.
4. Fix not pushing to docker unstable when using `build_from_cache` in dispatch workflow

### CODE REVIEW

**Review by quaark — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -43,6 +43,8 @@ MLRUN_CACHE_DATE ?= $(shell date +%s)
 # 2. add the --cache-from flag to the docker build
 # 3. docker tag and push (also) the (updated) cache image
 MLRUN_DOCKER_CACHE_FROM_TAG ?=
+# if MLRUN_NO_CACHE passed we don't want to use cache, this is mainly used for cleaner if statements
+MLRUN_USE_CACHE := $(if $(MLRUN_NO_CACHE),,true)
```
Comment (Hedingber) [inline Makefile:47]: The order I was trying to keep in the Makefile is that the first block of variables are ones that a user might want to override (therefore most of them are `?=`) and the second block is of computed variables (therefore most of them are `:=`)
Doesn't have to be part of this PR but probably this convention should be backed by a comment in the Makefile itself to help people understand where to put new things

File: dockerfiles/base/Dockerfile [hunk 2]
```diff
@@ -5,9 +5,9 @@ FROM quay.io/mlrun/python:${MLRUN_PYTHON_VERSION}-slim
 RUN apt-get update && \
     apt-get -s dist-upgrade | \
     grep "^Inst" | \
-    grep -i securi | \
+    grep -i "stable\|securi" | \
```
Comment (Hedingber) [inline dockerfiles/base/Dockerfile:8]: I guess we should "explore" (and leave in the PR description / comment on code) what are the possible types we're filtering out by this grep, and what is the meaning of filtering in the `stable` ones (security is pretty clear, stable is not)

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: dockerfiles/base/Dockerfile (hunk 2 above)
Comment (Tankilevitch) [reply dockerfiles/base/Dockerfile:8]: This is mainly used to upgrade all installed packages, nothing really special about it, it just that I found that stable is the common for most packages that we missed

**Review by Hedingber — APPROVED** [review]

**Pull request merged**